content/en/docs/tasks/configure-pod-container/security-context.md

```diff
@@ -184,6 +184,21 @@ This field has no effect on ephemeral volume types such as
 and [`emptydir`](/docs/concepts/storage/volumes/#emptydir).
 {{< /note >}}
 
+## Delegating volume permission and ownership change to CSI driver
+
+{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
+
+If you deploy a [Container Storage Interface (CSI)](https://github.com/container-storage-interface/spec/blob/master/spec.md)
+driver which supports the `VOLUME_MOUNT_GROUP` `NodeServiceCapability`, the
+process of checking and changing ownership and permissions based on the
+`fsGroup` specified in the `securityContext` will be performed by the CSI driver
+instead of Kubernetes, provided that the `DelegateFSGroupToCSIDriver` Kubernetes
+feature gate is enabled. Since Kubernetes doesn't perform the the permission
+check and change, `fsGroupChangePolicy` does not take effect and the driver
+decides the algorithm to apply the change.
```

Comment:
Hmm - it is not much for driver to decide the algorithm. The CSI capability explicitly states that fsgroup can only be applied as an mount option. So CSI driver must not use this field to perform any recursive chown/chmod stuff.

So, we should set the expectation right in this document with something like:
> Since Kubernetes doesn't perform any ownership or permission change if driver supports `VOLUME_MOUNT_GROUP` - it is expected that driver will mount the volume with provided fsgroup, resulting in volume being readable/writable by the fsgroup.

Reply:
updated